### PR TITLE
MOSIP-24939 Improved code coverage.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java [hunk 1]
```diff
@@ -95,10 +93,9 @@ public InputStream downloadRegistrationCentersByHierarchyLevel(String langCode,
 			List<String> name) throws ResidentServiceCheckedException, IOException, Exception {
 		logger.debug("DownLoadMasterDataService::downloadRegistrationCentersByHierarchyLevel()::entry");
 		ResponseWrapper<?> proxyResponseWrapper = proxyMasterdataService
-				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, REGISTRATION_CENTER_TEMPLATE_NAME);
+				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, this.env.getProperty("resident.template.registration.centers.ist"));
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java]: Make it as a constant

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java [hunk 2]
```diff
@@ -128,17 +125,16 @@ public InputStream downloadRegistrationCentersByHierarchyLevel(String langCode,
...
 	/**
-	 * download nearest registration centers
+	 * download the nearest registration centers
 	 */
 	public InputStream getNearestRegistrationcenters(String langCode, double longitude, double latitude,
 			int proximityDistance) throws ResidentServiceCheckedException, IOException, Exception {
 		logger.debug("DownLoadMasterDataService::downloadRegistrationCentersByHierarchyLevel()::entry");
 		ResponseWrapper<?> proxyResponseWrapper = proxyMasterdataService
-				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, REGISTRATION_CENTER_TEMPLATE_NAME);
+				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, this.env.getProperty("resident.template.registration.centers.list"));
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java]: Make it as a constant

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java [hunk 3]
```diff
@@ -182,7 +178,7 @@ public InputStream convertByteArrayToInputStream(byte[] bytes){
 	public InputStream downloadSupportingDocsByLanguage(String langCode) throws ResidentServiceCheckedException, IOException, Exception {
 		logger.debug("ResidentServiceImpl::getResidentServicePDF()::entry");
 		ResponseWrapper<?> proxyResponseWrapper = proxyMasterdataService
-				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, SUPPORTING_DOCS_TEMPLATE_NAME);
+				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, this.env.getProperty("resident.template.support-docs-list"));
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java]: Make it as a constant

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 4]
```diff
@@ -876,6 +876,9 @@ public void validateFromDateTimeToDateTime(LocalDateTime fromDateTime, LocalDate
 		} else if(fromDateTime != null && toDateTime == null) {
 			audit.setAuditRequestDto(EventEnum.getEventEnumWithValue(EventEnum.INPUT_INVALID, "toDateTime", request_service_history_api));
 			throw new InvalidInputException("toDateTime");
+		} else if(fromDateTime == null && toDateTime == null) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java:882]: Change the strings to constant.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java]: done.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 4 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java:882]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java (hunk 3 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/DownLoadMasterDataServiceImpl.java]: done.

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**